### ISSUE TITLE
Statistical Inference -> Hypothesis Testing has errors

### ISSUE THREAD

**Issue by hasenstein** [swirldev/swirl_courses#172]
See https://class.coursera.org/statinference-014/forum/thread?thread_id=90

Summary of the reports:

-------------------------------------------------------------------------------------------------------------
|=======================================                                             |  46%
| Here again is the plot to show this. The shaded portion represents 5% of the area under the
| curve and those X values in it are those which are at least 1.64 standard deviations
| greater than the mean. The probability of this is 5%. This means that if our observed mean
| fell in this area we would reject a true null hypothesis, that mu=mu_0, with probability
| 5%.

Error: Results must be all atomic, or all data frames
| Leaving swirl now. Type swirl() to resume.
-------------------------------------------------------------------------------------------------------------
|====================                                                                |  24%
| Here's a plot to show what we mean. The shaded portion represents 5% of the area under the
| curve and those X values in it are those for which the probability that X>C is 5%.

Error: Results must be all atomic, or all data frames
| Leaving swirl now. Type swirl() to resume.
-------------------------------------------------------------------------------------------------------------
The error comes at the question: "Here's a plot to show what we mean. The shaded portion represents 5% of the area under the curve and those X values in it are those for which the probability that X>C is 5%."

Error in scale_apply(layer_data, x_vars, scale_train, SCALE_X, panel$x_scales) : 
  | Leaving swirl now. Type swirl() to resume.
-------------------------------------------------------------------------------------------------------------
|===================                                                            |  24%
| Here's a plot to show what we mean. The shaded portion represents 5%
| of the area under the curve and those X values in it are those for
| which the probability that X>C is 5%.

Error: arguments imply differing number of rows: 596, 589
Leaving swirl now. Type swirl() to resume.
-------------------------------------------------------------------------------------------------------------

**Comment by asadowns** [comment]
+1

**Comment by reginaastri** [comment]
I am unable to replicate your problem. It appears to me to be an issue with ggplot2. Are you using a current version of that package?

**Comment by hasenstein** [comment]
I clicked "Update" in the package tab of RStudio not more than 1 or 2 weeks ago and updated all that there was to update (and had done so frequently before).

**Comment by asadowns** [comment]
R Version 3.2.0 ggplot2 1.0.1

**Comment by paolobudroni** [comment]
Same Problem at the same point. 
Doing Hypothesis Test at around 46% I got an unexpected exit from Swirl.  GGPLOT2 is 1.0.1 and R is 3.2.0
I have re-built all the environment but the error is still there.

**Comment by paolobudroni** [comment]
I have re-installed R 3.1.3 and finally it seems that all is working fine.

**Comment by mikebirdgeneau** [comment]
@paolobudroni I'm having the same type of errors when trying to generate ggplot2 plots from a Shiny application; I posted some of the details I'm seeing here in case it helps. http://stackoverflow.com/questions/30520575/intermittent-errors-with-ggplot2-segfault-memory-not-mapped

**Comment by cbologa** [comment]
For both errors at 24% and 46%, the problem seems to be with the last 2 commands in 

https://github.com/swirldev/swirl_courses/blob/master/Statistical_Inference/Hypothesis_Testing/conf_5pct.R

When repeatedly running those commands without the suppressWarnings() part, print(g) sometimes generates only a warning

```
Warning message:
In loop_apply(n, do.ply) :
  Removed 1411 rows containing missing values (position_stack).
```

and other times it also gives different errors:

```
Error: Results must be all atomic, or all data frames
In addition: Warning messages:
1: In loop_apply(n, do.ply) :
  Removed 1411 rows containing missing values (position_stack).
2: In loop_apply(n, do.ply) :
  Removed 1411 rows containing missing values (position_stack).
```

or

```
Error: arguments imply differing number of rows: 596, 589
In addition: Warning message:
In loop_apply(n, do.ply) :
  Removed 1411 rows containing missing values (position_stack).
```
Maybe somebody with better R skills than mine can rewrite those 5 lines of code so that there is no need to suppress warnings and hide errors this way.
 
My platform info:
```
> sessionInfo()
R version 3.2.0 (2015-04-16)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] installr_0.15.17 ggplot2_1.0.1    swirl_2.2.21    

loaded via a namespace (and not attached):
 [1] Rcpp_0.11.6      digest_0.6.8     crayon_1.3.0     MASS_7.3-40      bitops_1.0-6     grid_3.2.0       plyr_1.8.2      
 [8] gtable_0.1.2     magrittr_1.5     scales_0.2.4     httr_0.6.1       stringi_0.4-1    reshape2_1.4.1   testthat_0.10.0 
[15] labeling_0.3     proto_0.3-10     tools_3.2.0      stringr_1.0.0    munsell_0.4.2    RCurl_1.95-4.6   yaml_2.1.13     
[22] colorspace_1.2-6 memoise_0.2.1
```

**Comment by mikebirdgeneau** [comment]
See if this helps you at all:
https://groups.google.com/forum/m/#!topic/ggplot2/DCtDNFHgvb0

**Comment by anfe67** [comment]
Same thing here. I have tried all the solutions above but I am still stuck.

**Comment by seankross** [comment]
@anfe67 Are you sure you have the latest version of R, ggplot2, swirl, etc?

**Comment by anfe67** [comment]
Yes, positive. I have tried updating all packages twice. And I have also found a workaround to complete the lesson. The 3 scripts in my .../R/i686-pc-linux-gnu-library/3.2/swirl/Courses/Statistical_Inference/Hypothesis_Testing/ named conf_5pct.R, conf_5pct_left.R conf_5pct_both.R contain offending lines, which is surrounded by suppressWarnings: 

suppressWarnings(g <- g+ layer("area",mapping = aes(x=ifelse(-9<x & x<qnorm(.05,sd=2),x,NA)),
            geom_params=list(fill="red",alpha=.5)) )

I suppose that these draw the areas under the bell to highlight the tails.

**Comment by anfe67** [comment]
I have also found a way to correct this one, which works fine for me. There is an article on r-bloggers: http://www.r-bloggers.com/how-to-shade-under-a-normal-density-in-r/. It basically defines a very complete R function to draw a shaded Normal distribution and highlight parts of it. And it gives examples of use. It builds on this article http://www.fernandohrosa.com.br/en/P/shaded_areas_in_r/ which uses base R plots instead of ggplot2.
